src/Ledger/GovernanceActions.lagda

```diff
@@ -324,22 +324,31 @@ It represents how the \agdaboundEnactState changes when a specific governance ac
 {\small
 \begin{code}
   _⊢_⇀⦇_,ENACT⦈_ : EnactEnv → EnactState → GovAction → EnactState → Set where
-  Enact-NoConf    : ⟦ gid ⟧ᵉ ⊢ s ⇀⦇ NoConfidence ,ENACT⦈  record s { cc = nothing , gid }
-  Enact-NewComm   : ⟦ gid ⟧ᵉ ⊢ s ⇀⦇ NewCommittee new rem q ,ENACT⦈ let
-    old = maybe proj₁ ∅ᵐ (proj₁ (EnactState.cc s))
-    in record s { cc = just ((new ∪ᵐˡ old) ∣ rem ᶜ , q) , gid }
-  Enact-NewConst  : ⟦ gid ⟧ᵉ ⊢ s ⇀⦇ NewConstitution dh sh ,ENACT⦈  record s { constitution = (dh , sh) , gid }
-  Enact-HF        : ⟦ gid ⟧ᵉ ⊢ s ⇀⦇ TriggerHF v ,ENACT⦈  record s { pv = v , gid }
-  Enact-PParams   : ⟦ gid ⟧ᵉ ⊢ s ⇀⦇ ChangePParams up ,ENACT⦈
+
+  Enact-NoConf    : ⟦ gid , t ⟧ᵉ ⊢ s ⇀⦇ NoConfidence ,ENACT⦈  record s { cc = nothing , gid }
+
+  Enact-NewComm   :
+    let old = maybe proj₁ ∅ᵐ (proj₁ (EnactState.cc s))
+    in rem ⊆ dom (old ˢ)
+    ────────────────────────────────
+    ⟦ gid , t ⟧ᵉ ⊢ s ⇀⦇ NewCommittee new rem q ,ENACT⦈
+      record s { cc = just ((new ∪ᵐˡ old) ∣ rem ᶜ , q) , gid }
```

Comment:
Should we also check that the intersection of old and new is empty?

Reply:
Right now it looks like it's possible to add and remove the same CC in one transaction

Reply:
or why do we need to check this at all?

Reply:
There are two ways to reelect someone in the current design: put them in `old` and `new`, or just in `new`. Putting them in `old` and `new` is slightly different since it only works if someone is already on the CC, while the other option always works.

Since they are slightly different I'm inclined to keep both options, but we could also prevent the one with an intersection.

Reply:
Ah, sorry, putting someone in `old` and `new` always removes that member, so there's no point in putting them in `new`. Yeah, then we should prevent non-trivial intersections.

Reply:
Actually, you're making a good point with whether the `rem ⊆ dom (old ˢ)` check is necessary. Since this GA is hash protected, it's already bound to updating one particular state to another, so all this check does is that if the community votes for something that includes some garbage in `rem` it gets delayed forever, which is arguably bad. If we add sanity checks for proposals (which we might), then that check really needs to go there. So I'll remove it from here.